### PR TITLE
Speed up resample with kernel generation modification

### PR DESCRIPTION
**Pull request by skim0514**
Modification from pull request #2415 to improve resample. 

Benchmarked for a 89% time reduction, tested in comparison to original resample method.

### CODE REVIEW

**Comment by carolineechen** [comment]
cc @adefossez

File: torchaudio/functional/functional.py [hunk 1]
```diff
@@ -1445,31 +1444,33 @@ def _get_sinc_resample_kernel(
...
+    if resampling_method == "sinc_interpolation":
+        window = torch.cos(t * math.pi / lowpass_filter_width / 2) ** 2
+    else:
+        # kaiser_window
+        if beta is None:
+            beta = 14.769656459379492
+        beta_tensor = torch.tensor(float(beta))
+        window = torch.i0(beta_tensor * torch.sqrt(1 - (t / lowpass_filter_width) ** 2)) / torch.i0(beta_tensor)
+
+    t *= torch.pi
```
Comment (carolineechen) [inline torchaudio/functional/functional.py:1465]: we can leave it using math.pi

File: torchaudio/functional/functional.py [hunk 2]
```diff
@@ -1445,31 +1444,33 @@ def _get_sinc_resample_kernel(
...
-            window = torch.i0(beta_tensor * torch.sqrt(1 - (t / lowpass_filter_width) ** 2)) / torch.i0(beta_tensor)
-        t *= math.pi
-        kernel = torch.where(t == 0, torch.tensor(1.0).to(t), torch.sin(t) / t)
-        kernel.mul_(window)
-        kernels.append(kernel)
+    t = torch.arange(0, -new_freq, -1, dtype=dtype)[:, None, None] / new_freq + idx
+    t *= base_freq
+    t = t.clamp_(-lowpass_filter_width, lowpass_filter_width)
 
     scale = base_freq / orig_freq
```
Comment (carolineechen) [inline torchaudio/functional/functional.py:1454]: scale can be moved down closer to where it's actually being used, after the window parts

**Comment by adefossez** [comment]
Looking good to me :)

**Review by nateanl — APPROVED** [review]
LGTM

File: torchaudio/functional/functional.py [hunk 3]
```diff
@@ -1445,31 +1444,32 @@ def _get_sinc_resample_kernel(
...
-            window = torch.i0(beta_tensor * torch.sqrt(1 - (t / lowpass_filter_width) ** 2)) / torch.i0(beta_tensor)
-        t *= math.pi
-        kernel = torch.where(t == 0, torch.tensor(1.0).to(t), torch.sin(t) / t)
-        kernel.mul_(window)
-        kernels.append(kernel)
+    t = torch.arange(0, -new_freq, -1, dtype=dtype)[:, None, None] / new_freq + idx
+    t *= base_freq
+    t = t.clamp_(-lowpass_filter_width, lowpass_filter_width)
+
+    # we do not use built in torch windows here as we need to evaluate the window
```
Comment (carolineechen) [inline torchaudio/functional/functional.py:1454]: there's a second line to this comment missing

**Review by carolineechen — APPROVED** [review]
missing the second line to the comment but otherwise looks good!

**Pull request closed without merging**